### ISSUE TITLE
Make TaskCloseable implement AutoCloseable directly

### ISSUE THREAD

**Issue by carl-mastrangelo** [perfmark/perfmark#207]
Currently it extends Closeable directly but that's due to being stuck on Java 6.

**Comment by szarnekow** [comment]
Since this now requires Java 8, AutoClosable is an option, isn't it?

**Comment by carl-mastrangelo** [comment]
Yes.    I think this may be safe to change, but I would need to think more about this.   I had keo StringFunction around due to ABI compatibility, but I dont think anyone access that type
